CRM/Utils/Check/Component/Timestamps.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2017                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2017
+ */
+class CRM_Utils_Check_Component_Timestamps extends CRM_Utils_Check_Component {
+
+  const DOCTOR_WHEN = 'https://github.com/civicrm/org.civicrm.doctorwhen';
+
+  /**
+   * Check that the timestamp columns are populated. (CRM-20958)
+   *
+   * @return array<CRM_Utils_Check_Message>
```

Comment:
It means "an array in which the elements are instances of class `CRM_Utils_Check_Message`".

Personally, I go a bit bonkers reading `@return array` because it doesn't really sanything.

AFAIK, `php.net` hasn't blessed a notation for array typing, but there a few references for comparison:
 * It's similar to Java's notation for generics (eg `Vector<Message>`, `HashMap<String,Message>`).
 * It matches [Hack's array notation](https://docs.hhvm.com/hack/collections/introduction). (Hack is PHP derivative with stronger typing.)
 * PHP-FIG has  [a draft spec](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#abnf) for docblocks. It mentions both C-style arrays (`Message[]`) as well Java-style generics (`Collection<Message>`).
 * In my local copy of PHPStorm, it provides drilldown for C-style arrays (`Message[]`) but not Java-style generics.
 * I'm not particularly consistent. Sometimes it seems meaningful to convey even more information, so I use really verbose descriptor like:

```
  @return array
     Array(string $msgId => CRM_Utils_Check_Message $msg).
```

I guess standardizing/documenting/cleaning-up would be good, but that should be probably be a separate project.